martin/src/cog/image.rs

```diff
@@ -0,0 +1,234 @@
+use std::fs::File;
+use std::io::BufWriter;
+use std::path::Path;
+
+use martin_tile_utils::TileCoord;
+use tiff::decoder::{Decoder, DecodingResult};
+
+use super::CogError;
+use crate::{MartinResult, TileData};
+
+/// Image represents a single image in a COG file. A tiff file may contain many images.
+/// This type contains several useful information and method for taking tiles from the image.
+#[derive(Clone, Debug)]
+pub struct Image {
+    /// IFD(Image file directory) number.
+    /// An IFD contains information about the image, as well as pointers to the actual image data.
+    pub ifd: usize,
+    /// How many tiles in a row of this image
+    pub across: u32,
+    // How many tiles in a column of this image
```

Comment:
Otherwise I think there is a verb missing ^^

```suggestion
    /// Number of tiles in a row of this image
    pub across: u32,
    // Number of tiles in a column of this image
```